### ISSUE TITLE
Elasticsearch Backend uses irrelevant where clauses

### ISSUE THREAD

**Issue by shredding** [torchbox/wagtail#1842]
I have a query like this:

```python
from wagtail.wagtailsearch.backends import get_search_backend
s = get_search_backend()

queryset = House.objects.filter(number_rooms__gt=3).filter(roof_shape=5).filter(price_ready__range=(5000, 250000))

result = s.search('Sample', queryset)
```

After hitting elasticsearch and correctly filtering out the correct primary keys, the backends extends the upper query here:

https://github.com/torchbox/wagtail/blob/master/wagtail/wagtailsearch/backends/elasticsearch.py#L415

This leads to an sql query like this:

```
SELECT fields FROM model  WHERE (number_rooms > 3.0 AND roof_shape_id = 5 AND price_ready" BETWEEN 5000 AND 250000 AND id IN (1033)
```

... hence applying totally irrelevant where clauses (the id is the primary key and would hence be sufficient since the filtering was already done by elasticsearch) that causes a query evalaution time of `0.039` on my machine.

Changing https://github.com/torchbox/wagtail/blob/master/wagtail/wagtailsearch/backends/elasticsearch.py#L415
 to 

```
        queryset = self.query.queryset.model.objects.filter(pk__in=pks)
```

brings it down by 10 times to `0.003`.

a) Is that a bug or a feature? If the backend was not able to apply some filters, wouldn't it be better to make that more explicit?

b) Is it possible to access the elasticsearch results directly? I have all the values there and do not want to hit the database.

**Comment by kaedroho** [comment]
I agree those where clauses are unneccessary, let's clean them up. It was implemented this way so things like prefech_related would be retained on the new queryset.

By the way, what database backend did you use in your benchmark?

a) I'd mark this as "cleanup/optimisation" since it's not adding anything new, or fixing broken functionality.
b) Interesting idea, that would certianly save a few milliseconds. Perhaps we could implement something like ``.values()`` or ``.values_list`` on search results?

**Comment by shredding** [comment]
Regarding b) I'd prefer a `raw=True` parameter or so on `.search()`, as i would expect `.values` to operate on the resulting database queryset.

**Comment by shredding** [comment]
... I used postgres.

**Comment by gasman** [comment]
This issue is still present. Confirmed by changing https://github.com/wagtail/wagtail/blob/e7e57cd29e4a4937aee8fff7f8e2103a7789ce89/wagtail/search/backends/elasticsearch7.py#L877-L878
to
```python
        # Find objects in database and add them to dict
        query = self.query_compiler.queryset.filter(pk__in=pks)
        print(query.query)
        for obj in query:
```

and running `Page.objects.filter(live=True).search("arepa")` in the shell - the resulting query includes the clause `WHERE ("wagtailcore_page"."live" AND "wagtailcore_page"."id" IN (37))`, showing that the redundant filter on `"wagtailcore_page"."live"` is still present.
